lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -25,33 +25,45 @@ def initialize(sandbox, project, target, umbrella_headers_by_dir = nil)
 
           # Creates the target in the Pods project and the relative support files.
           #
-          # @return [void]
+          # @return [TargetInstallationResult] the result of the installation of this target.
           #
           def install!
-            unless target.should_build?
-              add_resources_bundle_targets
-              return
-            end
-
             UI.message "- Installing target `#{target.name}` #{target.platform}" do
-              add_target
+              test_file_accessors, file_accessors = target.file_accessors.partition { |fa| fa.spec.test_specification? }
+
+              unless target.should_build?
+                # For targets that should not be built (e.g. pre-built vendored frameworks etc), we add a placeholder
+                # PBXAggregateTarget that will be used to wire up dependencies later.
+                native_target = add_placeholder_target
+                resource_bundle_targets = add_resources_bundle_targets(file_accessors)
+                return TargetInstallationResult.new(target, native_target, resource_bundle_targets)
+              end
+
               create_support_files_dir
+
+              native_target = add_target
+              resource_bundle_targets = add_resources_bundle_targets(file_accessors)
+
               if target.contains_test_specifications?
```

Comment:
I wonder if this conditional is necessary, eg if `add_test_targets` handles not having any test targets correctly?

Reply:
hm it actually will! let me see